### PR TITLE
Tracing: Show next/prev buttons when span filters are collapsed

### PR DESCRIPTION
**Pull request by joey-grafana**
**What is this feature?**

 Shows next/prev buttons when span filters are collapsed.

**Why do we need this feature?**

Improves UX so users can move between their span filter matches when the span filters is collapsed.

**Who is this feature for?**

Tracing users.

**Special notes for your reviewer:**

Screenshot 2023-07-04 at 07 18 00

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NewTracePageSearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NewTracePageSearchBar.tsx:1]: This is mostly just moving the file into the `/SearchBar/` folder

File: public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NewTracePageSearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+              Clear
+            </Button>
+            <div className={styles.matchesOnly}>
+              <Switch
+                value={showSpanFilterMatchesOnly}
+                onChange={(value) => setShowSpanFilterMatchesOnly(value.currentTarget.checked ?? false)}
+                label="Show matches only switch"
+              />
+              {/* TODO: fix keyboard a11y */}
+              {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NewTracePageSearchBar.tsx:92]: Temporary as being fixed in other PR.

File: public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NextPrevResult.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      <span className={styles.matches}>{spanFilterMatches ? matches : `${totalSpans} spans`}</span>
+      <div className={buttonEnabled ? styles.buttons : cx(styles.buttons, styles.buttonsDisabled)}>
+        <div
+          aria-label="Prev result button"
+          className={buttonClass}
+          onClick={(event) => prevResult(event, buttonEnabled)}
+          onKeyDown={(event) => prevResultOnKeyDown(event, buttonEnabled)}
+          role="button"
+          tabIndex={buttonEnabled ? 0 : -1}
+        >
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NextPrevResult.tsx:144]: Using `<div />` as this cannot be a nested button because it is to be used on the collapse bar header, which is already a button.

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx [hunk 4]
```diff
@@ -74,7 +74,7 @@ function getStyles(theme: GrafanaTheme2) {
       text-align: start;
       line-break: anywhere;
       margin-top: -${theme.spacing(0.25)};
-      margin-right: ${theme.spacing(4)};
+      margin-right: ${theme.spacing(6)};
```
Comment (matyax) [inline public/app/plugins/datasource/loki/components/LokiContextUi.tsx:77]: Is this an accidental change? Or how is this related with the rest of the PR?

Comment (joey-grafana) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:77]: This change was on purpose to make sure `LogContext` still looks great. Using a flex on the collapse bar so checked the components that use `Collapse` and `LogContext` looked like it could use a larger margin on the right.

Comment (matyax) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:77]: I see, thanks! How can I test this within your changes? Because using Loki in Explore there's nothing on the right of the query:

imagen

Comment (joey-grafana) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:77]: You should just be able to checkout this branch, build, run Loki and then open the `LogContext` as it's only adding extra margin right.

Doesn't look like you're seeing the revert icon in the far right (where the orange part is in your image). Is that missing from the DOM? (`Revert to initial log context query`)

Screenshot 2023-07-04 at 12 57 50

**Review by adrapereira — APPROVED** [review]
LGTM!

**Review by svennergr — APPROVED** [review]
Logs changes LGTM!
